### ISSUE TITLE
require-default-props fails to parse Flow's read only/covariant prop types

### ISSUE THREAD

**Issue by jsteiner** [yannickcr/eslint-plugin-react#1865]
Using Flow:

```js
type Props = {
  +top?: SpacingType,
}
```

errors:

> propType "+" is not required, but has no corresponding defaultProp declaration  react/require-default-props

The `+` prefix is throwing off the parser, causing it to look for a prop `+` rather than the prop `top`.

Using `eslint-plugin-react@^7.5.1`

**Comment by HenriBeck** [comment]
@ljharb Couldn't this also be a rule? 
Something like require-read-only-props when the project is using flow? 

Seems like a good practice to me.

**Comment by ljharb** [comment]
@HenriBeck I suppose it could be; but this issue is about the `default-props-match-prop-types` rule not understanding covariant syntax (and presumably contravariant as well).

**Comment by alexzherdev** [comment]
I think the title here is incorrect - the error message in the OP refers to `require-default-props`.
Anyway, the fix for this is coming 😄
